plugins/modules/systemd_facts.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2025, Marco Noce <nce.marco@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: systemd_facts
+short_description: Gather C(systemd) unit facts.
+description:
+  - This module gathers facts about systemd units (services, targets, sockets).
+  - It runs C(systemctl list-units) (or processes selected units) and collects properties
+    for each unit using C(systemctl show).
+  - Even if a unit has a loadstate of "not-found" or "masked", it is added to the fact,
+    but only with the minimal properties ( name, loadstate, activestate, substate).
+  - When O(select) is used, the module first checks if the unit exists. If not, the unit
+    is recorded with minimal information.
+  - The gathered information is added to ansible facts under the key systemd_units.
+version_added: "10.4.0"
+options:
+  select:
+    description:
+      - When set to V(true), only the units specified in O(unitname) are processed.
+    type: bool
+    default: false
+  unitname:
+    description:
+      - List of unit names to process when O(select) is V(true).
+    type: list
+    elements: str
+    default: []
```

Comment:
Why not simply remove the default and remove the option `select`, and let the module act on whether this option is `None` or a list?